cirq/api/google/v2/device.proto

```diff
@@ -0,0 +1,81 @@
+syntax = "proto3";
+
+package cirq.api.google.v2;
+
+option java_package = "com.google.cirq.api.google.v2";
+option java_outer_classname = "CalibrationProto";
+option java_multiple_files = true;
+
+// This contains information about a device that includes the
+// qubits on the device, supported gates, connections, and timing.
+// This message specifies information that is needed when sending a
+// Program message to the device.
+message DeviceSpecification {
+      // A list of allowed gatesets for programs submitted to this processor
+      // Language.gate_set should be one of these values to be valid.
+      repeated GateSet allowed_gate_sets = 1;
+
+      // A list of allowed ids for qubits within the Program.
+      // Any programs with ids not in this list will be rejected.
+      // If empty, all qubit values are allowed (e.g. in a simulator)
+      repeated string allowed_qubits = 2;
+
+      // For each qubit id, information about the qubit,
+      // including a list of qubits in its neighborhood
+      // (i.e. a two-qubit operation between these qubits is legal)
+      map<string, QubitInformation> qubit_information = 3;
+
+      // Additional recommendations, caveats, and soft requirements that
+      // are advice to users of the device, specified in English text
+      // For instance, "All Z gates are converted to VirtualZ gates".
+      string developer_recommendations = 4;
+}
+
+message GateSet {
+      // A list of allowed gatesets for programs submitted to this processor
+      // Language.gate_set should be one of these values to be valid.
+   string gate_set_name = 1;
+
+   repeated GateDefinition allowed_gates = 2;
+}
+
+message GateDefinition {
+
+    // A list of allowed ids for gates within the Program.
+    // Any programs with ids not in this list will be rejected.
+    string gate_name = 1;
+
+    // The number of qubits (targets) that can be specified for this gate.
+    // If unset or set to zero, any number of qubits is allowed.
+    int32 number_of_qubits = 2;
```

Comment:
If you have a 3-qubit gate, how would you specify which sets of qubits it can be applied to? For two qubit gates this is encoded in the neighbor information, but not clear how to do this for number_of_qubits > 2.

Reply:
Changed to Target